typescript/types.ts

```diff
@@ -174,6 +174,47 @@ export type Prompt = {
  */
 export type Output = ExecuteResult | Error;
 
+/**
+ * The output type of the result from executing a prompt.
+ * We use this the kind field to determine how to parse it.
+ */
+export type OutputData =
+  | {
+      kind: "string" | "file_uri" | "base64";
+      value: string;
+    }
+  /**
+   * The following is based off of ChatCompletionMessageToolCall
+   * from openai/types/chat and is used for general tool calls.
+   */
+  | {
+      kind: "tool_calls";
+      value: {
+        id: string;
```

Comment:
nit: should we make this optional? Not sure if everyone would have an id

Reply:
Or make it nullable -- string | null

Reply:
sg I'll make it optional